test/DynamoCoreTests/CodeBlockNodeTests.cs

```diff
@@ -262,6 +263,53 @@ public void Defect_MAGN_1045()
             Assert.AreEqual(1, codeBlockNode.OutPorts.Count);
         }
 
+        [Test]
+        public void CodeBlockConnectionsRemainAfterUndo()
+        {
+            RunCurrentModel();
+            // Create the initial code block node.
+            var codeBlockNode1 = CreateCodeBlockNode();
+            UpdateCodeBlockNodeContent(codeBlockNode1,"3;");
+
+            // Create the second code block node.
+            var codeBlockNode2 = CreateCodeBlockNode();
+            UpdateCodeBlockNodeContent(codeBlockNode2, "x;");
+
+            //connect them
+            this.CurrentDynamoModel.ExecuteCommand(
+                new DynamoModel.MakeConnectionCommand(codeBlockNode1.GUID, 0, PortType.Output,
+             DynamoModel.MakeConnectionCommand.Mode.Begin));
+            this.CurrentDynamoModel.ExecuteCommand(
+                new DynamoModel.MakeConnectionCommand(codeBlockNode2.GUID, 0, PortType.Input,
+                DynamoModel.MakeConnectionCommand.Mode.End));
+            RunCurrentModel();
+
+            var oldPos = codeBlockNode2.X;
+            var oldinputPortGuid = codeBlockNode2.InPorts[0].GUID;
+            var oldConnectorGuid = codeBlockNode2.InPorts[0].Connectors[0].GUID;
+
+            //move the second node.
+            this.CurrentDynamoModel.ExecuteCommand(
+                new DynamoModel.UpdateModelValueCommand(codeBlockNode2.GUID, "Position", "300.0;300.0"));
+            Assert.AreEqual(300, codeBlockNode2.X);
+            //undo it.
+            this.CurrentDynamoModel.CurrentWorkspace.Undo();
+            Assert.AreEqual(oldPos, codeBlockNode2.X);
+
+            //TODO not sure this is required behavior...
+            //assert that after undo the port and connector have the same GUIDs
+            //Assert.AreEqual(oldinputPortGuid, codeBlockNode2.InPorts[0].GUID);
+            //Assert.AreEqual(oldConnectorGuid, codeBlockNode2.InPorts[0].Connectors[0].GUID);
+
+
+            RunCurrentModel();
```

Comment:
how does undo change the value of this? cbn created - executed - moved from x to 300 -undo - moved to x. I am not sure whether this will change the value of cbn? or is this because the cbn is recreated in undo and the execution must be called one more time?

Reply:
I'm just using the x value as a way to check that the undo actually happened. - When a node property change is undone - the entire node is deserialized and recreated, thats whats causing the bug so we do an undo to position which would previously break the connectors.

Reply:
Is RunCurrentModel running / executing the graph? Why it is required after an Undo operation?

Reply:
I want to make sure that the node's value after a run is correct, this proves the node's connector still exists and is not pointing to an orphan port.